src/js/ripple/amount.js

```diff
@@ -342,8 +343,12 @@ Amount.prototype._check_limits = function() {
   if (this._value.isNaN() || this._value.isZero()) {
     return this;
   }
-  if (!this._is_native) {
-    var absval = this._value.absoluteValue();
+  var absval = this._value.absoluteValue();
+  if (this._is_native) {
+    if (absval.greaterThan(Amount.bi_xrp_max)) {
+      throw new Error('Exceeding max value of ' + Amount.bi_xrp_max.toString());
+    }
+  } else {
     if (absval.lessThan((new BigNumber(Amount.min_value)).absoluteValue())) {
       throw new Error('Exceeding min value of ' + Amount.min_value);
```

Comment:
Yeah, I was thinking the same thing. I was going to wait because I also want to unexpose all the bi_* symbols so that dependencies aren't tied to our internal bignumber library. But I think it's worth making this part of the change now.